go/x509util/x509util.go

```diff
@@ -0,0 +1,576 @@
+var (
+	OidExtensionArc                        = asn1.ObjectIdentifier{2, 5, 29} // id-ce RFC5280 s4.2.1
+	OidExtensionSubjectKeyId               = asn1.ObjectIdentifier{2, 5, 29, 14}
+	OidExtensionKeyUsage                   = asn1.ObjectIdentifier{2, 5, 29, 15}
+	OidExtensionExtendedKeyUsage           = asn1.ObjectIdentifier{2, 5, 29, 37}
+	OidExtensionAuthorityKeyId             = asn1.ObjectIdentifier{2, 5, 29, 35}
+	OidExtensionBasicConstraints           = asn1.ObjectIdentifier{2, 5, 29, 19}
+	OidExtensionSubjectAltName             = asn1.ObjectIdentifier{2, 5, 29, 17}
+	OidExtensionCertificatePolicies        = asn1.ObjectIdentifier{2, 5, 29, 32}
+	OidExtensionNameConstraints            = asn1.ObjectIdentifier{2, 5, 29, 30}
+	OidExtensionCRLDistributionPoints      = asn1.ObjectIdentifier{2, 5, 29, 31}
+	OidExtensionIssuerAltName              = asn1.ObjectIdentifier{2, 5, 29, 18}
+	OidExtensionSubjectDirectoryAttributes = asn1.ObjectIdentifier{2, 5, 29, 9}
+	OidExtensionInhibitAnyPolicy           = asn1.ObjectIdentifier{2, 5, 29, 54}
+	OidExtensionPolicyConstraints          = asn1.ObjectIdentifier{2, 5, 29, 36}
+	OidExtensionPolicyMappings             = asn1.ObjectIdentifier{2, 5, 29, 33}
+	OidExtensionFreshestCRL                = asn1.ObjectIdentifier{2, 5, 29, 46}
+
+	OidExtensionAuthorityInfoAccess = asn1.ObjectIdentifier{1, 3, 6, 1, 5, 5, 7, 1, 1}
+	OidExtensionSubjectInfoAccess   = asn1.ObjectIdentifier{1, 3, 6, 1, 5, 5, 7, 1, 11}
+
+	// RFC 6962 s3.1
+	OidExtensionCTPoison = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 3}
+	// RFC 6962 s3.3
+	OidExtensionCTSCT = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 2}
+)
+
+//////////////////////////////////////
+
+// OidForStandardExtension indicates whether oid identifies a standard extension.
+// Standard extensions are listed in RFC 5280 (and other RFCs).
+func OidForStandardExtension(oid asn1.ObjectIdentifier) bool {
+	if oid.Equal(OidExtensionSubjectKeyId) ||
+		oid.Equal(OidExtensionKeyUsage) ||
+		oid.Equal(OidExtensionExtendedKeyUsage) ||
+		oid.Equal(OidExtensionAuthorityKeyId) ||
+		oid.Equal(OidExtensionBasicConstraints) ||
+		oid.Equal(OidExtensionSubjectAltName) ||
+		oid.Equal(OidExtensionCertificatePolicies) ||
+		oid.Equal(OidExtensionNameConstraints) ||
+		oid.Equal(OidExtensionCRLDistributionPoints) ||
+		oid.Equal(OidExtensionIssuerAltName) ||
+		oid.Equal(OidExtensionSubjectDirectoryAttributes) ||
+		oid.Equal(OidExtensionInhibitAnyPolicy) ||
+		oid.Equal(OidExtensionPolicyConstraints) ||
+		oid.Equal(OidExtensionPolicyMappings) ||
+		oid.Equal(OidExtensionFreshestCRL) ||
+		oid.Equal(OidExtensionSubjectInfoAccess) ||
+		oid.Equal(OidExtensionAuthorityInfoAccess) ||
+		oid.Equal(OidExtensionCTPoison) ||
+		oid.Equal(OidExtensionCTSCT) {
+		return true
+	}
+	return false
+}
+
+// OidInExtensions indicates whether the extension identified by oid is present in extensions.
+func OidInExtensions(oid asn1.ObjectIdentifier, extensions []pkix.Extension) (int, bool) {
```

Comment:
Could you add to the godoc comment for this function to say what the bool returned represents?  I don't know what the 'Critical' field in extensions if for, and it may be obvious to someone who does, but indicating that that field is what is being returned would be helpful if someone was just looking at the godoc for this package, and not the actual code.

In fact, explaining in the comment that the int returned is the number of extensions in extensions that have oid, and bool is if any of them are critical would be good.  From the comment, I initially made the assumption that the bool indicated whether the extension identified by oid is present in extensions, and wasn't sure what the int was for (both of which were not quite right!)